### DISCUSSION TITLE
Pairing problem(timeout)

### DISCUSSION THREAD

**Discussion by mflash20** [HomeSpan/HomeSpan#819] [Q&A]
so I've just started with homespan and lets just say its fantastic.     
but i haven't been able to pair the device to HomeKit. it finds the device and when i enter the setup key . it says adding to home then after a few seconds it says operation timed out.   please help me solve this ,I'd appreciate it so much. Thanks

**Comment by HomeSpan** [comment]
Can you provide more details on exactly what is happening.  Best to start with a completely **unmodified** version of Tutorial Example 1.  To ensure a clean build, type 'E' into the CLI to erase all stored data, load Example 1, use the 'W' CLI command to enter your WiFi Credentials, and then after HomeSpan reboots, type 'L2' into the CLI to turn on full logging messages.  Then try pairing using the default set-up code of 46637726.

**Reply by mflash20** [reply]
even after doing that i still get the timeout message in HomeKit  .this is the log.
thanks for your dedication.

21:54:27.161 -> *** Log Level set to 2
21:54:27.161 -> 
21:54:28.265 -> Trying to connect to home.  Waiting 4 sec...
21:54:28.446 -> WiFi Connected!  IP Address = 192.168.1.26
21:54:28.446 -> 
21:54:28.446 -> Starting MDNS...
21:54:28.446 -> 
21:54:28.446 -> HostName:      HomeSpan-75B05A571EA0.local:80
21:54:28.446 -> Display Name:  HomeSpan LightBulb
21:54:28.446 -> Model Name:    HomeSpan-ESP32
21:54:28.446 -> Setup ID:      HSPN
21:54:28.446 -> 
21:54:28.446 -> Starting HAP Server on port 80 supporting 14 simultaneous HomeKit Controller Connections...
21:54:28.446 -> 
21:54:28.446 -> 
21:54:28.446 -> DEVICE NOT YET PAIRED -- PLEASE PAIR WITH HOMEKIT APP
21:54:28.446 -> 
21:54:49.244 -> =======================================
21:54:49.244 -> ** Client #0 Connected: (26 sec) 192.168.1.131 on Socket 2/16
21:54:49.244 -> 
21:54:49.244 -> <<<<<<<<< 192.168.1.131 <<<<<<<<<
21:54:49.244 -> POST /pair-setup HTTP/1.1
21:54:49.244 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:54:49.244 -> Content-Length: 6
21:54:49.244 -> Content-Type: application/pairing+tlv8
21:54:49.244 -> ------------ END BODY! ------------
21:54:49.244 -> STATE(1) 01
21:54:49.244 -> METHOD(1) 00
21:54:49.244 -> ------------ END TLVS! ------------
21:54:49.244 -> In Pair Setup #0 (192.168.1.131)...Found <M1>.  Expected <M1>.
21:54:49.360 -> 
21:54:49.360 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:54:49.360 -> HTTP/1.1 200 OK
21:54:49.360 -> Content-Type: application/pairing+tlv8
21:54:49.360 -> Content-Length: 409
21:54:49.360 -> 
21:54:49.360 -> SALT(16) C23E9006497B6A129CD54BFDD9FAD207
21:54:49.360 -> PUBKEY(384) 4C194F82FE264A9DDC1B85B97B23B124E498D014E6BC33BD9B6C9C0FA838BC66D2E695D0F9965B89F92E83F16430F8FD4C1C21C2CC3851D176E8D41FDF61941CD6B2E6DC7DE79AF86C97F19C844BDCF093C52E8FD8CAFC4EAFB718A2B28C2044556E339C991F60999976B84C56AB859C486EF546D404F45766DB038E1C461382A52E24AC18491B5ACB972B088811D5E272CF9C4C44C24782F8D42D5E2CB3CBEF8917BD09A8B4C466800A8F9EFDEA36D95C138F90E376983545B87DAD8B23279A3C7456F078C5A193CEA2E5EA86EEDF2CA15D29A41C7EE0E541AC9CB8D6239CD81381ECA84C0CA3BBEF80980208B94231D84D091F9F941AB377F88C0676E0C08C2831519BA342C004BC0F0B0C64E090F77850F561BE50C883365206ACA028A15F248907B5CEA32E30DA05CAE717DBB2227CC7266C2C15A8F9B91D36FB98FE9A065920E644FFA79B0BE154B08E0B3DC99D5DF1BE743545A7BE02834706CD5FA56A9CC67998102B9A6D79776070F074695A1BC7E5F3A49C6E1C8D53B7E1337268B0
21:54:49.398 -> STATE(1) 02
21:54:49.435 -> ------------ SENT! --------------
21:54:49.435 -> 
21:55:06.383 -> <<<<<<<<< 192.168.1.131 <<<<<<<<<
21:55:06.384 -> POST /pair-setup HTTP/1.1
21:55:06.384 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:06.384 -> Content-Length: 457
21:55:06.384 -> Content-Type: application/pairing+tlv8
21:55:06.384 -> ------------ END BODY! ------------
21:55:06.384 -> PROOF(64) C0FEF05F947F6962B96AD0009E4B6A0322CE4E1BF6428D1085D6F5B3749AD15D03BAF0236AD3D7D83FBD1DAA5419D43D86F150554410CBF1F09929AB74C88DC5
21:55:06.384 -> PUBKEY(384) 2532C330E51C3C1F385286C11F97CC2CCB4419AEC41134961F7F9EE343D6609F4E8BD6489CE621352F1DF1E4DF070FE49EBA2628A14372A0FB61724110523BAE78909DE459056EC4D727A0BF9AB5975D5B033BF795AE53C943ADAA6E7DEFAA4D34E11AA60D74014690AC3EA914188CC396424935E05CDA9370C5AA113F04CBD29550CE5E1270919A2A04E9B9CAA7C2518DB3346F256343C05E19A48F4ED4F76125B7CE30FC8E8302B0EADBB355310EE6C328BB53E1C8AB03C154EA9813C26ED140B35A43D4D3D5D375D1217C6843D39A692881DE7071BC6093223269FF80B6EC1A791B765DAD546F4237F30FC95165E10B8CD0AC14A927F3837B8E05A52E654FDFDD8B73963D11F6B95B1FB05FB31C175DB67E63FC08E0A63BD6C7EA785049CC08F0963848F96F9EBA2B6FCDC4C0D1E3B27F54C4732617E8D6E980C43E0F57C7D7C178E40602F1D0174AEA01690A36C2068939151B04A1120DD9E161894A62C27BD38FC87CE2EBFA261C54D9986270A9A5BBE7C6ABCA2E12F11EB2423F637525
21:55:06.430 -> STATE(1) 03
21:55:06.430 -> ------------ END TLVS! ------------
21:55:06.430 -> In Pair Setup #0 (192.168.1.131)...Found <M3>.  Expected <M3>.
21:55:06.564 -> 
21:55:06.564 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:06.564 -> HTTP/1.1 200 OK
21:55:06.564 -> Content-Type: application/pairing+tlv8
21:55:06.564 -> Content-Length: 69
21:55:06.564 -> 
21:55:06.564 -> PROOF(64) 5F143C18364CD4CC0BCCE0F814D3AA6FC105E8D0FF2845061E3DE3470235A5295595D5449C8C2F36201131EE6E572222DE8EBC0812D0DA74CC6120E4F8687C22
21:55:06.564 -> STATE(1) 04
21:55:06.600 -> ------------ SENT! --------------
21:55:06.600 -> 
21:55:06.770 -> <<<<<<<<< 192.168.1.131 <<<<<<<<<
21:55:06.770 -> POST /pair-setup HTTP/1.1
21:55:06.770 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:06.770 -> Content-Length: 159
21:55:06.770 -> Content-Type: application/pairing+tlv8
21:55:06.770 -> ------------ END BODY! ------------
21:55:06.770 -> STATE(1) 05
21:55:06.770 -> ENC.DATA(154) C0F67A43872DB316A47A1A5B118ADBD79E20666819E97BBE62DD05C8E0C0EEE9B22AA959F513AF8BF218F754C28BCB19079E35B0D8E7C3BE885D2B51C2876F982F3FDD13BBA18DD9A4780FFC0D5FE1F4316170970F19D81A33A286FF86ABE5537894610F93CF73462192461E02E77F6A7A9155B294BEDDFE97FB2AC3EADF9127666D75A315168B3A0884F57FB3F58366521B51BE681E7C9A65E7
21:55:06.850 -> ------------ END TLVS! ------------
21:55:06.850 -> In Pair Setup #0 (192.168.1.131)...Found <M5>.  Expected <M5>.
21:55:06.850 -> ------- DECRYPTING SUB-TLVS -------
21:55:06.850 -> SIGNATURE(64) D99B7E6B21843EA74E4920BCF4400425075B0D3C1F0124562001B7C47690154970C0A914462B8E91AE778A291CE46BDB1ABEBB6A47915644C7DE232DDD1CF206
21:55:06.850 -> PUBKEY(32) 64B471276A3C721E9DD406A95DE0274C019D5ED1F5B323B5BD8A743785DDF9FA
21:55:06.850 -> IDENTIFIER(36) 46383932443338302D363742462D343131432D383741312D353241393342334439453032
21:55:06.850 -> ---------- END SUB-TLVS! ----------
21:55:06.850 -> 
21:55:06.850 -> *** Added Controller: F892D380-67BF-411C-87A1-52A93B3D9E02 (admin)
21:55:06.850 -> 
21:55:06.929 -> ------- ENCRYPTING SUB-TLVS -------
21:55:06.929 -> PUBKEY(32) 729D49D40A50F51CC7D7DE8B35B4519D53331697133C59382FBBB0825703B7AF
21:55:06.929 -> IDENTIFIER(17) 37353A42303A35413A35373A31453A4130
21:55:06.929 -> SIGNATURE(64) C375A7EA8A3F7619B8C85AA37B2F29912340B783219C9B795CFF02F4B2852A7C046E185056D803D2201CDB8C0E9CEE743FEF84E3113CACEA4BEEFD30900BCA0D
21:55:06.929 -> ---------- END SUB-TLVS! ----------
21:55:06.929 -> 
21:55:06.929 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:06.929 -> HTTP/1.1 200 OK
21:55:06.929 -> Content-Type: application/pairing+tlv8
21:55:06.929 -> Content-Length: 140
21:55:06.929 -> 
21:55:06.929 -> ENC.DATA(135) B223A72129847A0DECD7D7474BFDC2215300C400A47F83A52B137C215276AC3FED077ECDB874CB3C0AADB28A295483181C65D65DE2C80E097ACEC3FEC545CAD01C5BA02616F358B1E7F481A69AA23A7BFC7A07F884985435188DE62D407E549EEE663289E109749D6D69FD11BCDE61D510B3455422FA01C96BD5DE8A9402FBAE278810936A2A3C
21:55:06.929 -> STATE(1) 06
21:55:06.929 -> ------------ SENT! --------------
21:55:06.929 -> 
21:55:06.929 -> *** ACCESSORY PAIRED! ***
21:55:06.929 -> 
21:55:07.107 -> =======================================
21:55:07.107 -> ** Client #0 Connected: (44 sec) 192.168.1.131 on Socket 3/16
21:55:07.107 -> 
21:55:07.107 -> <<<<<<<<< 192.168.1.131 <<<<<<<<<
21:55:07.107 -> POST /pair-verify HTTP/1.1
21:55:07.107 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:07.107 -> Content-Length: 37
21:55:07.107 -> Content-Type: application/pairing+tlv8
21:55:07.107 -> ------------ END BODY! ------------
21:55:07.107 -> PUBKEY(32) 39771157309EEE2334BBD2BFE661AD11323E65FDD02AFDE17E32CC524A5E4B54
21:55:07.107 -> STATE(1) 01
21:55:07.107 -> ------------ END TLVS! ------------
21:55:07.107 -> In Pair Verify #0 (192.168.1.131)...Found <M1>
21:55:07.107 -> ------- ENCRYPTING SUB-TLVS -------
21:55:07.107 -> SIGNATURE(64) 5D7E290365730D5444C177E726DD540FB4C2ECEA6CC62B57EF9F768D047732FC8508A8BDD8E316B4750AEF61472BFE040749EBBA30519A4A68FD0865F50EC90F
21:55:07.107 -> IDENTIFIER(17) 37353A42303A35413A35373A31453A4130
21:55:07.194 -> ---------- END SUB-TLVS! ----------
21:55:07.194 -> 
21:55:07.194 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:07.194 -> HTTP/1.1 200 OK
21:55:07.194 -> Content-Type: application/pairing+tlv8
21:55:07.194 -> Content-Length: 140
21:55:07.194 -> 
21:55:07.194 -> PUBKEY(32) F5D7E58EC4A2EC7275FE23C889BA8A29AF2D5351352CEFA943D710AC2D866D0E
21:55:07.194 -> STATE(1) 02
21:55:07.194 -> ENC.DATA(101) BEF6B8B31F4B08413EEE1158CACD7A96E88A43E28EA0967897F7FE64F865A8181C4620B2E2A67B8C4C738886B788E7FD35E882DB1CDDB616F0746EF8BA9CA59B96415388A536972A19F46ACCA4F4BB9936DB1DB97C931E6C4FD34C3CBCB6B24D94EF86CFA9
21:55:07.194 -> ------------ SENT! --------------
21:55:07.194 -> 
21:55:07.505 -> <<<<<<<<< 192.168.1.131 <<<<<<<<<
21:55:07.505 -> POST /pair-verify HTTP/1.1
21:55:07.505 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:07.505 -> Content-Length: 125
21:55:07.505 -> Content-Type: application/pairing+tlv8
21:55:07.505 -> ------------ END BODY! ------------
21:55:07.505 -> STATE(1) 03
21:55:07.505 -> ENC.DATA(120) 07C7F6715BFDD0096E1B3B9FCE17FF1F95EA0D27EF1880C5FD196590664CB444108AB0C5004EFA0086D92781BC644A006E236F958E2FDA0AE7416020119733AD5A5ABC0A098100459A0A1F56BAADB14FEBDC2F33C4B485F5C62549DA44ED891D1FE81708BDAC109206938AFBF45BFB101F8EA9B8E595602A
21:55:07.505 -> ------------ END TLVS! ------------
21:55:07.505 -> In Pair Verify #0 (192.168.1.131)...Found <M3>
21:55:07.553 -> ------- DECRYPTING SUB-TLVS -------
21:55:07.553 -> SIGNATURE(64) 97E1F70E8905F26B9AD8893E3A936D01F749C9AC97B94F394EF0D0613308F96ED92AC62398F0B76613C2FDA034D8D2019A3DDB832965D6D8DD20F9CC54407E0A
21:55:07.553 -> IDENTIFIER(36) 46383932443338302D363742462D343131432D383741312D353241393342334439453032
21:55:07.553 -> ---------- END SUB-TLVS! ----------
21:55:07.553 -> 
21:55:07.553 -> *** Verifying session with Controller ID: F892D380-67BF-411C-87A1-52A93B3D9E02...
21:55:07.600 -> 
21:55:07.600 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:07.600 -> HTTP/1.1 200 OK
21:55:07.600 -> Content-Type: application/pairing+tlv8
21:55:07.600 -> Content-Length: 3
21:55:07.600 -> 
21:55:07.600 -> STATE(1) 04
21:55:07.600 -> ------------ SENT! --------------
21:55:07.600 -> 
21:55:07.600 -> *** SESSION VERIFICATION COMPLETE *** 
21:55:07.600 -> 
21:55:07.600 -> <<<< #### 192.168.1.131 #### <<<<
21:55:07.600 -> GET /accessories HTTP/1.1
21:55:07.600 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:07.600 -> ------------ END BODY! ------------
21:55:07.600 -> In Get Accessories #0 (192.168.1.131)...
21:55:07.600 -> 
21:55:07.600 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:07.600 -> HTTP/1.1 200 OK
21:55:07.600 -> Content-Type: application/hap+json
21:55:07.600 -> Content-Length: 250
21:55:07.600 -> 
21:55:07.600 -> {"accessories":[{"aid":1,"services":[{"iid":1,"type":"3E","characteristics":[{"iid":2,"type":"14","format":"bool","perms":["pw"]}]},{"iid":3,"type":"43","characteristics":[{"iid":4,"type":"25","value":0,"format":"bool","perms":["pr","pw","ev"]}]}]}]}
21:55:07.646 -> -------- SENT ENCRYPTED! --------
21:55:07.646 -> 
21:55:14.668 -> <<<< #### 192.168.1.131 #### <<<<
21:55:14.668 -> POST /pairings HTTP/1.1
21:55:14.668 -> Host: HomeSpan\032LightBulb._hap._tcp.local
21:55:14.668 -> Content-Length: 44
21:55:14.668 -> Content-Type: application/pairing+tlv8
21:55:14.668 -> ------------ END BODY! ------------
21:55:14.668 -> IDENTIFIER(36) 46383932443338302D363742462D343131432D383741312D353241393342334439453032
21:55:14.668 -> STATE(1) 01
21:55:14.668 -> METHOD(1) 04
21:55:14.668 -> ------------ END TLVS! ------------
21:55:14.668 -> In Post Pairings #0 (192.168.1.131)...Remove...
21:55:14.668 -> 
21:55:14.668 -> >>>>>>>>>> 192.168.1.131 >>>>>>>>>>
21:55:14.668 -> HTTP/1.1 200 OK
21:55:14.668 -> Content-Type: application/pairing+tlv8
21:55:14.668 -> Content-Length: 3
21:55:14.668 -> 
21:55:14.668 -> STATE(1) 02
21:55:14.715 -> -------- SENT ENCRYPTED! --------
21:55:14.715 -> 
21:55:14.715 -> *** Removing Controller: F892D380-67BF-411C-87A1-52A93B3D9E02 (admin)
21:55:14.715 -> *** Terminating Client #0
21:55:14.715 -> That was last Admin Controller!  Removing any remaining Regular Controllers and unpairing Accessory
21:55:14.715 -> ** Disconnected Client #0  (51 sec)
21:55:14.715 ->

**Reply by HomeSpan** [reply]
According to the above log, the device is properly paired and verified, but then HomeKit requests that the controller be removed.  HomeSpan processes the request and unpairs the device.  I've not seen this before, nor can even venture a guess as to why HomeKit is trying to unpair right after pairing.  Are you using any other software, such as Home Assistant (HA)?  Are you using a HomeKit Hub (Apple TV or HomePod)?

**Reply by mflash20** [reply]
no. not using any other software. i've added tvs and other devices to HomeKit before and this is the first time i get the timout error.

**Reply by mflash20** [reply]
is there anyway to solve this ?

**Reply by HomeSpan** [reply]
what version of iOS are you using?  Also, have you recently added a commercial device using the same iPhone?

**Reply by mflash20** [reply]
I'm using the 17.4.1 version and no the last time i added a device was a year ago and that was a tv

**Reply by HomeSpan** [reply]
I'm on 17.4.1 and just test Example 1 with HomeSpan 1.9.0, and though HomeKit is requested to remove a controller, it is an unused ID, not the one that was paired.

Is there a commercial device you can test pairing with help narrow the scope of the problem (note if there is an issue with your configuration and you unpair an existing device, you may not be able re-pair it, so test with caution).

**Reply by mflash20** [reply]
SO I've used my phone and another phone but still the same problem. maybe the problem is with the home span version?

**Reply by HomeSpan** [reply]
for the test i meant try to pair a commercial device with the same phone.  

i assume you are using homespan 1.9.0?

**Reply by mflash20** [reply]
Yes i'm using 1.9 and it works fine with other devices like a sony tv

**Reply by HomeSpan** [reply]
To confirm, we are trying to test only the pairing part, not whether or not the Home App can operate a device.  When you say "it" works fine with other devices, I assume you mean that from your Home App you can control other devices.  But have you tried to do a fresh pairing with another device (after first unpairing it?).  Also, the device to test needs to be WiFi, not Ethernet, since connectivity is slightly different using the two methods.

**Reply by mflash20** [reply]
Just tried to add another device and no . i get the same error.     you said it cant pair with a home hub right? does apple tv count as a home hub?

**Reply by HomeSpan** [reply]
Okay, so this means the issue is not with HomeSpan but rather something about your network configuration.  You need to have a Home Hub, and an Apple TV counts if it's Gen 4 or greater.  If you have had success with pairing other devices in the past, but now things are not working, this suggests that your Apple TV may not have been updated with the latest version of TVOS.  Make sure all Apple products on your network have the latest operating systems --- this can be the issue.

**Comment by G-D-D-971** [comment]
I have a similar problem, if I change the pairing code / setup ID It's impossible to pair the accessory with HomeKit… (I try with the 
example 04 by adding :
```
homeSpan.setPairingCode("26728027");
homeSpan.setQRID("JT92");
```
Just after `void setup()`…
for unique result I receive that :
![Screenshot 2024-03-26 at 18 13 55](https://github.com/HomeSpan/HomeSpan/assets/124715754/6b1455b9-b074-46c9-9a5b-85db923c95e3)
Or timeout…
P.S : also try to change the pairing code with the CLI with `S 24558975` same problem, and same result for setup code… 

And another problem is the status update (between all devices in the HomeKit network) don't work anymore.
![IMG_0203](https://github.com/HomeSpan/HomeSpan/assets/124715754/07ac0883-a858-4606-98fe-571d812f12a7)
![IMG_2930](https://github.com/HomeSpan/HomeSpan/assets/124715754/0b143b98-4cfc-40b9-a209-60fbdc3021f3)
I used homespan 1.9.0

**Reply by HomeSpan** [reply]
This seems unrelated to the original topic of this thread - the problem @mflash20 is having is not related to the Setup Code.  

With regard to above, HomeKit is telling you that you don't have a Home Hub connected.  HomeSpan will not function properly without a Home Hub (neither will any commercial devices as of about iOS 16.3).

**Comment by mflash20** [comment]
Thank you so much for the help.
Thanks. It really means a lot
